modules/yieldmoSyntheticInventoryModule.js

```diff
@@ -98,35 +115,20 @@ function serialize(dataObj) {
   return str.join('&');
 }
 
-function processResponse(response) {
-  let responseBody;
+function processResponse(res) {
+  let parsedResponseBody;
   try {
-    responseBody = JSON.parse(response.responseText);
+    parsedResponseBody = JSON.parse(res.responseText);
   } catch (err) {
-    throw new Error(`${MODULE_NAME}: response body is not valid JSON`);
+    throw new Error(`${MODULE_NAME}: response is not valid JSON`);
   }
-  if (response.status !== 200 || !responseBody.data || !responseBody.data.length || !responseBody.data[0].ads || !responseBody.data[0].ads.length) {
-    throw new Error(`${MODULE_NAME}: NOAD`);
+  if (res && res.status === 204) {
```

Comment:
Previously the idea was to check if there is an actual ad in the response. now there are checks for the status and error code. will this checks cover all the cases when the response does not have an ad?

Reply:
I guess there are 4 cases: 
1. 200 with ad (data.ads is no empty and no error_code).
2. 200 with empty data.ads and error_code
3. 204 which is a valid response too
4. >300 including 500

So validation covers 2, 3. 4 covers by ajax library, 1 will let code go ahead.